### ISSUE TITLE
#3184: Viewers: Tile Viewer: Scroll reset and rendering issues after adding new viewer

### ISSUE THREAD

**Issue by LesiaPavlenko** [datagrok-ai/public#3184]
- [ ] When adding a Tile Viewer and scrolling to the middle, then adding another viewer, the scroll position resets, and the Tile Viewer fails to render. To trigger rendering, manual actions like scrolling or moving elements are required.
- [ ] Molecule images flicker and are rendered twice, which should not happen. This issue needs further investigation to ensure smooth rendering.

![tile_viewer_issue](https://github.com/user-attachments/assets/0b8d40d8-14aa-4d8d-89e9-29c49c6bd754)

**Comment by dnillovna** [comment]
This issue has been mirrored in Jira: https://reddata.atlassian.net/browse/GROK-17081

**Comment by katerina-stepanova** [comment]
The same issue can be reproduced for NX custom viewer.

**Comment by drizhina** [comment]
Hi @katerina-stepanova. Does the NX custom viewer use DG virtual item view?

**Comment by katerina-stepanova** [comment]
@drizhina Our viewer is inherited from DG.JsViewer. Not sure what is the 'virtual item view', however our developers say they haven't used this feature, at least not explicitly.
Could you share a link to its documentation if there is any?

**Comment by LesiaPavlenko** [comment]
added to 1.23.0

**Comment by drizhina** [comment]
The issue with tile viewer has been fixed.
After some investigation, we found the core issue, which is most probably the same in your custom viewers.
The Dock manager of datagrok (the system that docks different viewers together, enables resizing and dragging) physically rearranges the 'viewer nodes' in HTML to position different items in most convinient way. when this happens, the native scroll of rearranged nodes is reset. We fixed it for 'virtual item view', which is the backend of tiles viewer, and also Legend. If your custom viewers were using Virtual item view, then the issue will be fixed on your side as well.
